internal/evaluator/conftest_evaluator.go

```diff
@@ -190,25 +213,59 @@ func (c conftestEvaluator) Evaluate(ctx context.Context, inputs []string) ([]out
 		result.Warnings = warnings
 		result.Failures = failures
 
-		runResults[i] = result
+		results = append(results, CheckResult{CheckResult: result})
 	}
 
-	results = append(results, runResults...)
+	// set successes, these are not provided in the Conftest results, so we
+	// reconstruct these from the parsed rules, any rule that hasn't been
+	// touched by adding metadata must have succeeded
+
+	// TODO see about multiple resuts, somehow; using results[0] for now
+	if l := len(rules); l > 0 {
+		results[0].Successes = make([]output.Result, 0, l)
+	}
+
+	for code, rule := range rules {
```

Comment:
I don't follow how this is filtering out rules that did fail/warn.

Reply:
Oh that's confusing! Unrelated to this PR it seems

Reply:
We have parsed AST of rego rules, we don't know which ones passed and which ones
failed, so we need to delete the ones that we know failed (or warninged), that
should in theory leave only the ones that succeeded. So very much part of this PR.